cmds/boot/boot.go

```
@@ -0,0 +1,403 @@
+	}
+	return (blkDevices)
+}
+
+// checkForBootableMBR is looking for bootable MBR signature 
+// Current support is limited to Hard disk devices and USB devices
+func checkForBootableMBR(path string) (bool,error) {
+	var sig uint16
+	f, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	if err := binary.Read(io.NewSectionReader(f, signatureOffset, 2), binary.LittleEndian, &sig); err != nil {
+		return false,err
+	}
+	if sig != bootableMBR {
+		return false, nil
+	}
+	return true,nil	
+}
+
+// getDevicePartList returns all devices attached to a specific name like /dev/sdaX where X can move from 0 to 127
+// FIXME no support for devices which are included into subdirectory within /dev
+func getDevicePartList(path string) []string {
+	var returnValue []string
+	files, err := ioutil.ReadDir("/dev/")
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), path) {
+			// We shall not return full device name
+			if file.Name() != path {
+				// We shall check that the remaining part is a number
+				returnValue = append(returnValue, file.Name())
+			}
+		}
+	}
+	return returnValue
+}
+
+// getSupportedFilesystem returns all block file system supported by the linuxboot kernel
+func getSupportedFilesystem() []string {
+	var returnValue []string
+	var err error
+	file, err := os.Open("/proc/filesystems")
+	check(err)
+	scanner := bufio.NewScanner(file)
+	if err = scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		if fields[0] == "nodev" {
+			continue
+		}
+		if fields[0] != "" {
+			returnValue = append(returnValue, fields[0])
+    		}
+	}
+
+	err=file.Close()
+	check(err)
+	return returnValue
+
+}
+
+// mountEntry tries to mount a specific block device
+func mountEntry(path string, supportedFilesystem []string) bool {
+	var returnValue bool
+	var err error
+	err=syscall.Mkdir("/u-root", 0777)
```

use ioutil.TempDir, I'm getting errors as this file exists. 
Or use os.IsExist(err) and ignore the error if so.

---

Should be fixed in latest commit